### ISSUE TITLE
Problems with testing a few rules

### ISSUE THREAD

**Issue by oxyzero** [alexgarrett/violin#20]
So I was writing some basic unit tests for some rules and I got a few problems.

```php
public function testAlnumRule()
{
        $this->v->validate(['username' => 'violin-tests_alnum'], ['username' => 'alnum']);
        $this->assertFalse($this->v->valid());

        $this->v->validate(['username' => 'violin'], ['username' => 'alnum']);
        $this->assertTrue($this->v->valid());

        $this->v->validate(['username' => 'violin-tests@alnum'], ['username' => 'alnum']);
        $this->assertFalse($this->v->valid());
}
```
In the second assertion, the validation should return true, instead it returns false. I debugged and saw that the rule returns 1 (and I casted it into boolean) and it still gave me false.

Can anyone try to unit test this rule and explain to me why my test is failing?

Also, is the ActiveUrl rule correct? I made a few changes into the rule itself when I was doing some tests with it. My tests are the following:
```php
public function testActiveUrl()
{
        $this->v->validate(['url' => 'http://www.google.com'], ['url' => 'activeUrl']);
        $this->assertTrue($this->v->valid());

        $this->v->validate(['url' => 'google.com'], ['url' => 'activeUrl']);
        $this->assertTrue($this->v->valid());

        $this->v->validate(['url' => md5('test')], ['url' => 'activeUrl']);
        $this->assertFalse($this->v->valid());
}
```

The problem is, the rule returns !checkdnsrr(), which in the first assertion is equal to true (because www.google.com is invalid), the second assertion returns true as it should, and then, when the md5 is not a valid url at all, it returns true in the third assertion as well.

So my changes to the Violin\Rules\ActiveUrlRule were:
```php
public function run($name, $value)
    {
        $url = str_replace(array('http://', 'https://', 'ftp://', 'www.'), '', strtolower($value));
        $url = idn_to_ascii($url);
        return checkdnsrr($url);
    }
```

The idn_to_ascii function is to convert the url into ascii based on this example on php.net:
```php
var_dump(checkdnsrr('ñandu.cl', 'A')); // returns false
var_dump(checkdnsrr(idn_to_ascii('ñandu.cl'), 'A')); // returns true
```

**Comment by oxyzero** [comment]
Alright, so this morning I look into it and I figured out why testAlnumRule() was returning false when it should return true. The reason for this is that everytime Violin performs a validation it never clears the error array. What this means is that if we want to perform multiple validations with one instance of Violin, we can't. So what I did was in Violin\Violin->validate() I clean up the errors array. (Check PR #19)

But what does everyone think about the changes to the Active URL rule?
